Integrations/integration-cisco-ise.yml

```diff
@@ -0,0 +1,543 @@
+
+            update_json = json.loads(update_result.text)
+
+            updated_fields_dict_list = update_json.get('UpdatedFieldsList', {}).get('updatedField',[])
+
+            if len(updated_fields_dict_list) > 0:
+                updated_fields_string = ' the new custom fields are: ' + json.dumps(updated_fields_dict_list[0].get('newValue'))
+            else:
+                updated_fields_string = ", but the fields that you've tried to update already had that specific value"
+
+            return 'Successfully updated endpoint %s' % endpoint_id + updated_fields_string
+
+        except Exception as e:
+            raise Exception("Exception: Failed to update endpoint {}: ".format(endpoint_id) + str(e))
+
+    def update_endpoint_group_command():
+        """
+        corresponds to 'cisco-ise-update-endpoint-group' command. Updates endpoint status
+        """
+
+        endpoint_group_name = demisto.args().get('groupName')
+        endpoint_group_id = demisto.args().get('groupId')
+
+        if not endpoint_group_name and not endpoint_group_id:
+            return 'Please enter either group id or group name'
+
+        if endpoint_group_name and not endpoint_group_id:
+            endpoint_group_data = get_endpoint_id(None, endpoint_group_name).get('SearchResult', {})
+            if endpoint_group_data.get('total',0) < 1:
+                return 'No endpoints were found. Please make sure you entered the correct group name'
+
+            endpoint_group_id = endpoint_group_data.get('resources')[0].get('id')
+
+        endpoint_id = demisto.args().get('id')
+        endpoint_mac_address = demisto.args().get('macAddress')
+
+        if endpoint_mac_address and not is_mac(endpoint_mac_address):
+            return "Please enter a valid mac address"
+
+        if not endpoint_id and not endpoint_mac_address:
+            return 'Please enter either endpoint id or endpoint mac address'
+
+        if endpoint_mac_address and not endpoint_id:
+            endpoint_id = get_endpoint_id(endpoint_mac_address).get('SearchResult', {}).get('resources', [])[0].get('id', None)
+
+        endpoint_details = get_endpoint_details(endpoint_id)
+
+        if "ERSEndPoint" not in endpoint_details:
+            return 'Failed to get endpoint %s' % endpoint_id
+
+        try:
+            endpoint_details['ERSEndPoint']['groupId']= endpoint_group_id
+            update_result = update_endpoint_by_id(endpoint_id, endpoint_details)
+
+            # Create result
+            msg = "Endpoint " + endpoint_id + " updated successfully" if update_result.status_code == 200 else "Update failed for endpoint " + endpoint_id + ", got the following response: " + \
+                    json.dumps(json.loads(update_result.text).get('ERSResponse',{}).get('messages',[]))
+            result = [{ "Update status": msg }]
+
+        except Exception as e:
+            raise Exception("Exception: Failed to update endpoint {}: ".format(endpoint_id) + str(e))
```

Comment:
Non existing endpoint ID raises the following: `Exception: expected string or buffer`